pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -21,6 +22,8 @@ import (
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
 )
 
+const finalizer = "mongodbatlas/finalizer"
```

Comment:
I think it's better to move the CR finalizer const to the finalizer.go file. And use it in all CR.